packages/big-design/src/components/Tree/hooks/spec.ts

```diff
@@ -0,0 +1,347 @@
+        preventDefault,
+        stopPropagation,
+      } as unknown) as React.KeyboardEvent<HTMLLIElement>,
+      { id: '1', isExpanded: false, isSelectable: true, hasChildren: false, value: 1 },
+    );
+
+    expect(preventDefault).toHaveBeenCalledTimes(1);
+    expect(stopPropagation).not.toHaveBeenCalled();
+    expect(onFocus).toHaveBeenCalledWith('0');
+  });
+
+  test('arrow up on first node does nothing', () => {
+    visibleNodes = ['0', '1'];
+    const { result } = renderHook(() => useTreeKeyEvents({ onFocus, onSelect, onToggle, nodeMap, visibleNodes }));
+
+    result.current(
+      ({
+        currentTarget: 1,
+        target: 1,
+        key: 'ArrowUp',
+        preventDefault,
+        stopPropagation,
+      } as unknown) as React.KeyboardEvent<HTMLLIElement>,
+      { id: '0', isExpanded: false, isSelectable: true, hasChildren: false, value: 0 },
+    );
+
+    expect(preventDefault).toHaveBeenCalledTimes(1);
+    expect(stopPropagation).not.toHaveBeenCalled();
+    expect(onFocus).toHaveBeenCalledWith('0');
+  });
+
+  describe('arrow right (has children)', () => {
+    test('if expanded, focuses on next visible node', () => {
+      visibleNodes = ['0', '1'];
+      const { result } = renderHook(() => useTreeKeyEvents({ onFocus, onSelect, onToggle, nodeMap, visibleNodes }));
+
+      result.current(
+        ({
+          currentTarget: 1,
+          target: 1,
+          key: 'ArrowRight',
+          preventDefault,
+          stopPropagation,
+        } as unknown) as React.KeyboardEvent<HTMLLIElement>,
+        { id: '0', isExpanded: true, isSelectable: true, hasChildren: true, value: 0 },
+      );
+
+      expect(preventDefault).toHaveBeenCalledTimes(1);
+      expect(stopPropagation).not.toHaveBeenCalled();
+      expect(onFocus).toHaveBeenCalledWith('1');
+    });
+
+    test('if collapsed, expands node', () => {
+      const { result } = renderHook(() => useTreeKeyEvents({ onFocus, onSelect, onToggle, nodeMap, visibleNodes }));
+
+      result.current(
+        ({
+          currentTarget: 1,
+          target: 1,
+          key: 'ArrowRight',
+          preventDefault,
+          stopPropagation,
+        } as unknown) as React.KeyboardEvent<HTMLLIElement>,
+        { id: '0', isExpanded: false, isSelectable: true, hasChildren: true, value: 0 },
+      );
+
+      expect(preventDefault).toHaveBeenCalledTimes(1);
+      expect(stopPropagation).not.toHaveBeenCalled();
+      expect(onToggle).toHaveBeenCalledWith('0', false);
+    });
+  });
+
+  describe('arrow left', () => {
+    test('if expanded, collapses node', () => {
+      visibleNodes = ['0', '1'];
+
+      const { result } = renderHook(() => useTreeKeyEvents({ onFocus, onSelect, onToggle, nodeMap, visibleNodes }));
+
+      result.current(
+        ({
+          currentTarget: 1,
+          target: 1,
+          key: 'ArrowLeft',
+          preventDefault,
+          stopPropagation,
+        } as unknown) as React.KeyboardEvent<HTMLLIElement>,
```

Comment:
🍹  could we use `new KeyboardEvent(...)` instead of manually constructing the object, to get rid of the `as` castings? never tried this before so i'm not sure if it'd work

Reply:
Good idea, let me try it out!

Reply:
So I checked it out, not sure the types are going to work cause `React.KeyboardEvent !== KeyboardEvent`. The react type provides a generic element to be passed in, while the native `KeyboardEvent` doesn't and relies on `EventTarget` as it's "element".